ESLint/lombiq-base.js

```diff
@@ -0,0 +1,160 @@
+module.exports = {
+    "extends": "./eslint-base.js",
+
+    // Here we can define our own rules overriding the base rules
+    "rules": {
+        // Lombiq rules
+        "max-len": [
+            "error",
```

Comment:
Az összes lehet warning, ha utána ez nem probléma és ugyanúgy MSBuild warning tud az egészből lenni.

Reply:
Airbnb-nél a nagyja error, lényegében nem számít, hogy mit írunk ki, hogy warning vagy error, mert az MSBuild máshogy fogja meg a dolgot, de konzisztensebb úgy ha error marad

Reply:
A warning azért lenne jobb (de tényleg csak akkor, ha a dolgot többi része ugyanúgy működik), mert akkor a .NET analyzerekkel lenne konzisztens. És az, hogy ezek warningot fejlesztés közben, jobb a produktivitáshoz: ha írsz valami kódot, mert próbálgatsz, akkor nem baj, ha nem tökéletes éppen akkor, attól még a Gulp buildnek mennie kellene. A lényeg, hogy ugye az egész végén legyen már hibátlan, azt pedig a CI build kikényszeríti, localban pedig ugyanúgy látod az összes hibát warningként.

Reply:
Akkor írjak át minden errort az airbnb baseben IS warningra? Ugye a java ott van

Reply:
Nem ott, itt.

Reply:
De ugye a nagy része az ott van, és ha csak itt lenne minden warning az elég furán venné ki magát, hogy a ruleok 1/8~ része warning maradék meg error

Reply:
Tehát amit mondok, hogy az **összes** rule-nak warningnak kellene lennie. Ehhez le kell másolnunk kb. az Airbnb fájlt? Vagy lehet csak az egyes rule-oknak a leveljét változtatni? Vagy még inkább, az összeset egyben? Utóbbi úgy tűnik lehet: https://stackoverflow.com/questions/46517832/how-to-extend-airbnb-eslint-but-with-warnings-instead-of-errors

Reply:
Most megnéztem amit linkeltél, az egy elavult package, az újnál már [itt](https://github.com/webpack-contrib/eslint-webpack-plugin#emitwarning) azt írja, hogy minden esetben kiírja a warningot, azt nem hogy az errort is átalakítja, de a legegyszerűbb az lenne, hogy a `base-rules`-ban és `lombiq-rules`-ban replacelni minden "error"-t "warning"-ra szerintem "kézzel"

Reply:
De igazából ezzel az a helyzet, hogy ez csak esztétikai kérdés, mert az MSBuild ha error ha warning megfogja és csinál egy MSBuild warningot, hogy találta az ESLint problémát, ami ugye localban csak warning CI-n megbukik, és az egyetlen amit számít, hogy a VS beépített ESLint-je milyen messaget ír ki az error list-be, de annál sem számít ugye ez nem fog megállítani semmilyen buildet (localban)

Reply:
OK az is. Ha lehet csak a levelt változtatni, ahol más configot nem akarunk egy rule-ról, akkor legyen az, ha nem lehet, akkor ez van.

Reply:
Nem csak esztétikai kérdés, mert localban nem akarjuk a buildet ezekkel megakasztani, a Gulp buildet sem. PG-nél most pont ez a probléma: https://teams.microsoft.com/l/message/19:525a1a3656c845ae92c5e2c0d3617cc7@thread.skype/1613654278764?tenantId=959ede5a-38e9-44e6-adc8-85553beb8220&groupId=d6a5d775-6fb9-41dc-a7bb-481d4f884046&parentMessageId=1613654278764&teamName=Lombiq%20Developers&channelName=%C3%81ltal%C3%A1nos&createdTime=1613654278764

Reply:
Azóta ez variálva lett és nem akasztja meg az error a buildet

Reply:
Csak ugye az még nincs dev-en fent, de `js-targets.js`-ből kikerült a `failOnError()` amit felvált most már az MSBuild warning

Reply:
Jah. Akkor valóban csak esztétikai (kommunikációs) kérdés.

Reply:
Akkor maradjon ahogy van, vagy legyen warning levelre állítva minden ami error-on van? Ez a rule configján nem változtat csak warningot printel error helyett outputba

Reply:
Legyen warning. Tényleg konzisztensek akarunk maradni (és meghagyni annak a lehetőségét, hogy később majd valami olyan probléma lehessen, amit localban is azonnal javítani kell, mint az ESLint exception).

Reply:
Most ugye az eslint-base.js-ben átállítottad az összeset warningra. Bár ettől tényleg warning lesz minden, de ha frissíteni akarjuk az Airbnb repoból a fájlt, akkor ismét mindent meg kell változtatnunk. Ezért írtam ide a lombiq-base-hez, mert ezt innen kellene megoldanunk. Lásd innen: https://github.com/Lombiq/Gulp-Extensions/pull/11#discussion_r577867157